control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -201,6 +205,20 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager, create
 	r.ec2Client, r.awsSession = GetEC2Client()
 
 	r.registerComponents(hcp)
+
+	uidInput := os.Getenv(controlplaneoperator.DefaultSecurityContextUIDEnvVar)
+	if uidInput == "" {
+		r.Log.Info("DEFAULT_SECURITY_CONTEXT_UID is not set. This should never happen, unless you are running a HO which doesn't support this CPO")
+		r.DefaultSecurityContextUID = 0
```

Comment:
Should this fallback to 1001 which is what we have today?

Reply:
we are falling back to that already via [support/controlplane-component/defaults.go](https://github.com/openshift/hypershift/pull/6520/files#diff-a194f3bdb82070261e10c3429532c0fbc35ec6b199238ccaece0d509fb788863)

This can only be reached if running this cpo with an older HO. I only added this code path because that's the case for the ho upgrade test.